pkg/vere/io/http.c

```diff
@@ -302,6 +304,71 @@ _http_heds_from_noun(u3_noun hed)
   return hed_u;
 }
 
+/* _http_req_is_auth(): returns c3y if rec_u contains a valid auth cookie
+*/
+static c3_o
+_http_req_is_auth(u3_hfig* fig_u, h2o_req_t* rec_u)
+{
+  //  try to find a cookie header
+  //
+  h2o_iovec_t coo_u = {NULL, 0};
+  {
+    //TODO  http2 allows the client to put multiple 'cookie' headers,
+    //      runtime should support that once eyre does too.
+    ssize_t hin_i = h2o_find_header_by_str(&rec_u->headers, "cookie", 6, -1);
+    if ( hin_i != -1 ) {
+      coo_u = rec_u->headers.entries[hin_i].value;
+    }
+  }
+
+  //  if there is no cookie header, it can't possibly be authenticated
+  //
+  if ( NULL == coo_u.base ) {
+    return c3n;
+  }
+  //  if there is a cookie, see if it contains a valid auth token
+  //
+  else {
+    c3_c* key_c = fig_u->key_c;
+    c3_c  val_c[128];
+    c3_y  val_y = 0;
+    c3_y    i_y = 0;
+    c3_y    j_y = 0;
+
+    //  step through the cookie string
+    //
+    while (i_y < coo_u.len) {
+      //  if we found our key, read the value
+      //
+      if (key_c[j_y] == '\0' && coo_u.base[i_y] == '=') {
+        i_y++;
+        while (i_y < coo_u.len && coo_u.base[i_y] != ';') {
+          val_c[val_y] = coo_u.base[i_y];
+          val_y++;
+          i_y++;
+        }
+        break;
+      }
+      //  keep reading the key as long as it matches
+      //
+      else if (coo_u.base[i_y] == key_c[j_y]) {
+        j_y++;
+      }
+      else {
+        j_y = 0;
+      }
+      i_y++;
+    }
+
+    u3_noun aut = u3kdi_has(
+      u3k(fig_u->ses),
+      u3i_bytes(val_y, (c3_y*)val_c));
```

Comment:
Nit: this is a little hard to read. I think one of the following would be easier to read:
```c
u3_noun aut = u3kdi_has(u3k(fig_u->ses),
                        u3i_bytes(val_y, (c3_y*)val_c));
// or
u3_noun aut = u3kdi_has(
    u3k(fig_u->ses),
    u3i_bytes(val_y, (c3_y*)val_c)
);